tests/TestCase/Network/ResponseTest.php

```diff
@@ -2096,4 +2096,65 @@ public function testLocation()
         $this->assertNull($response->location('http://example.org'), 'Setting a location should return null');
         $this->assertEquals('http://example.org', $response->location(), 'Reading a location should return the value.');
     }
+
+    /**
+     * Test get protocol version.
+     *
+     * @return void
+     */
+    public function getProtocolVersion()
+    {
+        $response = new Response();
+        $version = $response->getProtocolVersion();
+        $this->assertEquals('1.1', $version);
+    }
+    /**
+     * Test with protocol.
+     *
+     * @return void
+     */
+    public function testWithProtocol()
+    {
+        $response = new Response();
+        $version = $response->getProtocolVersion();
+        $this->assertEquals('1.1', $version);
+        $response2 = $response->withProtocolVersion('1.0');
+        $version = $response2->getProtocolVersion();
+        $this->assertEquals('1.0', $version);
+        $version = $response->getProtocolVersion();
+        $this->assertEquals('1.1', $version);
+        $this->assertNotEquals($response, $response2);
+    }
+    /**
+     * Test with protocol.
+     *
+     * @return void
+     */
+    public function testWithStatusCode()
+    {
+        $response = new Response();
+        $statusCode = $response->getStatusCode();
+        $this->assertEquals(200, $statusCode);
+        $response2 = $response->withStatus(404);
+        $statusCode = $response2->getStatusCode();
+        $this->assertEquals(404, $statusCode);
+        $statusCode = $response->getStatusCode();
+        $this->assertEquals(200, $statusCode);
+        $this->assertNotEquals($response, $response2);
+    }
+
+    /**
+     * Test get reason phrase.
+     *
+     * @return void
+     */
+    public function testGetReasonPhrase()
+    {
+        $response = new Response();
+        $reasonPhrase = $response->getReasonPhrase();
+        $this->assertNull($reasonPhrase);
+        $response = $response->withStatus(404);
+        $reasonPhrase = $response->getReasonPhrase();
+        $this->assertEquals('Not Found', $reasonPhrase);
+    }
```

Comment:
We'll need tests for the other header methods. I know they are implemented via a trait that also has tests, but if we ever change that it would be good to have tests covering the behavior we promised people.